cip-tbd/cip-tbd.md

```diff
@@ -0,0 +1,503 @@
+
+The ground truth for the dApp API is maintained in the Splice Wallet repository in a machine-readable form. The methods and events listed in this document represent those that **MUST** be implemented by Wallet Providers and provide a convenient summary of the state of the API at the time of writing for readers of the CIP.
+
+The dApp API exists in two variants, each targeting a different deployment model and interaction pattern:
+
+1. **[Synchronous dApp API](https://github.com/hyperledger-labs/splice-wallet-kernel/blob/main/api-specs/openrpc-dapp-api.json)** (aka Sync dApp API) – Intended for clients with direct access to a Wallet, such as browser extensions or local Wallets. Methods and events execute in a request–response fashion, enabling immediate ledger reads, transaction preparation, and signing flows.
+
+
+2. **[Asynchronous dApp API](https://github.com/hyperledger-labs/splice-wallet-kernel/blob/main/api-specs/openrpc-dapp-remote-api.json)** (aka Async dApp API) – Designed for server-side Wallets or Wallets deployed as services that cannot support blocking calls or direct synchronous interactions. This variant decomposes operations requiring user interaction into multi-phase workflows, often exposing `userUrl`’s to allow users to complete actions asynchronously (for example, by interacting with a UI) while the client process continues to wait for an event to be emitted (as a result of the user interaction).
+
+Clients such as frontends that work against the sync dApp API **SHOULD** also work against a server exposing the async dApp API. This ensures that applications can support both local and remote Wallet configurations without changing core business logic.
+We expect frontends to do so by implementing the sync dApp API by relaying calls to an async dApp API server. See the [reference dApp SDK implementation](#references) for a drop-in solution.
+
+#### Synchronous dApp API
+
+The **Synchronous dApp API** variant is intended for clients that have direct access to a Wallet, such as browser extensions, desktop applications, or mobile Wallets. In this model, methods and events are executed in a standard request–response fashion, allowing dApps to perform ledger reads, transaction preparation, and signing operations immediately and deterministically.
+
+Wallet Providers implementing this variant **MUST** support all listed methods and events, ensuring consistent behavior and error semantics across deployments. This variant is the canonical choice for end-user applications, providing a familiar synchronous programming model and minimizing latency between user actions and dApp responses.
+
+Overview of supported methods:
```

Comment:
Is the splice-wallet-kernel going to be updated to support all methods? I do not see them as of this commit, https://github.com/hyperledger-labs/splice-wallet-kernel/blob/e13742492b9a8758e5b03c8d0f5822b8a2dadd79/sdk/dapp-sdk/src/provider/request.ts#L13-L105

Should we follow this CIP draft instead?